### ISSUE TITLE
Raster tiles are not rendered correctly for some elements

### ISSUE THREAD

**Issue by utagawal** [maptiler/tileserver-gl#698]
On version 4.3.x the rendering between vector and raster is not aligned for some elements

![RASTER](https://user-images.githubusercontent.com/16464382/211063613-377f4643-1ec7-4f08-967d-7b8895eb3cac.gif)

Missing rendering this color overlay style :

```
{
  "id": "mtb-no-access-global",
  "type": "line",
  "source": "openmaptiles",
  "source-layer": "transportation",
  "minzoom": 13,
  "filter": [
    "all",
    ["in", "class", "path", "track"],
    [
      "any",
      [
        "all",
        ["==", "bicycle", "no"],
        ["in", "mtb_scale", "5", "6"]
      ],
      [
        "all",
        ["==", "bicycle", "no"],
        ["!has", "mtb_scale"]
      ],
      [
        "all",
        ["==", "access", "no"],
        ["!=", "bicycle", "yes"],
        ["!has", "mtb_scale"]
      ]
    ]
  ],
  "layout": {
    "visibility": "visible",
    "line-cap": "round",
    "line-join": "bevel"
  },
  "paint": {
    "line-dasharray": [1],
    "line-opacity": 0.4,
    "line-color": "rgba(255, 0, 0, 1)",
    "line-width": 6
  }
}
```

And footway :

```
{
  "id": "path-footway",
  "type": "line",
  "source": "openmaptiles",
  "source-layer": "transportation",
  "minzoom": 11,
  "filter": [
    "all",
    ["in", "class", "path"],
    ["!=", "subclass", "steps"],
    [
      "in",
      "subclass",
      "footway",
      " pedestrian"
    ]
  ],
  "layout": {
    "line-cap": "round",
    "line-join": "round",
    "visibility": "visible"
  },
  "paint": {
    "line-color": "rgba(255, 255, 255, 1)",
    "line-width": {
      "base": 1.55,
      "stops": [[6, 2], [20, 5]]
    },
    "line-opacity": 1,
    "line-gap-width": 0,
    "line-dasharray": [1]
  }
}
```

and also hillshading

Test can be done here :
* [raster](https://maps.utagawavtt.com/styles/utagawavtt/?raster#18/45.55305/4.63218)
* [vector test](https://maps.utagawavtt.com/styles/utagawavtt/#17/45.552365/4.632938)

**Comment by acalcutt** [comment]
Which one is right? Am I reading the style as it should be white like the raster map?

**Comment by acalcutt** [comment]
This would likely need to be fixed on the maplibre-native side. Has this work in any 4.x.x release or did it last work in 3.x?

**Comment by acalcutt** [comment]
Does using hex values when you specify colors in your style make any difference?

**Comment by utagawal** [comment]
> Does using hex values when you specify colors in your style make any difference?

I changed it to see if this would have any effect and it makes no difference.

> This would likely need to be fixed on the maplibre-native side. Has this work in any 4.x.x release or did it last work in 3.x?

Not sure, I just found out the issue on the current release

**Comment by acalcutt** [comment]
Thanks for testing. 

What I would like to try next is to see if I can replicate this in maplibre-native linux version, but following the "Using your own style/tiles" steps in https://github.com/maplibre/maplibre-gl-native/tree/main/platform/linux. If it happens there it will help me make a issue in the maplinbre-native repo

I tried a bit to do this earlier, but maplibre was getting a 403 on your relief layer, so I have to try and simplify it a bit
```
xvfb-run --server-args="-screen 0 1024x768x24" ./build/bin/mbgl-render --style https://maps.utagawavtt.com/styles/utagawavtt/style.json --output out.png -z17 -x45.55305 -y4.63218
[ERROR] {mbgl-render}[Style]: Failed to load tile 15/20530/15962=>17 for source relief-france-hd: HTTP status code 403
Error: HTTP status code 403
```

**Comment by utagawal** [comment]
I have removed the ip filtering on the relief and then it works for the hillshading :-), so that part is fixed :-)

**Comment by acalcutt** [comment]
With that change I was able to test. I ran this command
```
 xvfb-run --server-args="-screen 0 1024x768x24" ./build/bin/mbgl-render --style https://maps.utagawavtt.com/styles/utagawavtt/style.json --output out.png -z17 -y45.55305 -x4.63218 -w2000 -h1500
```

And it gave me back this image which looks like it has the same issue.
![out](https://user-images.githubusercontent.com/3792408/211124905-4be17b60-084b-49d9-b637-471b50145d30.png)

So this confirms it is either a maplibre native issue or a style issue. I should note it did give this one warning when i ran the command above/

`[WARNING] {mbgl-render}[ParseStyle]: line dasharray requires at least two elements`

I may run yot style though the [style-spec tools](https://github.com/maplibre/maplibre-gl-js/tree/main/src/style-spec) and see if it finds any style issues.
